doc/misc/system-state.7.rst

```diff
@@ -0,0 +1,40 @@
+..
+    Copyright Contributors to the libdnf project.
+
+    This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+    Libdnf is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 2 of the License, or
+    (at your option) any later version.
+
+    Libdnf is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+
+.. _systemstate_misc_ref-label:
+
+##############
+ System state
+##############
+
+Description
+===========
+
+The DNF5 system state consists of several TOML files located in the /usr/lib/sysimage/libdnf5/ directory. DNF5 uses the system state to:
+
+    1. Store the reasons why each installed package was added to the system. The reasons can be "user" for packages that the user explicitly asked DNF5 to install, "dependency" and "weak dependency" for packages pulled in as dependencies of another package, "group" for packages installed by a group, or "external" for packages installed by another tool (e.g. rpm).
+
+    2. Track installed groups and packages installed by these groups.
+
+    3. Track installed environmental groups.
+
+
+Recovering from a corrupted system state
+========================================
+
+If the system state files become corrupted, simply back up and remove the corrupted TOML file mentioned in the error message. It will be regenerated during the next successful DNF transaction.
```

Comment:
"it will be regenerated" leaves an impression that files can be removed without any loss. That's not true. Could you add a warning: "The regenerated file will be missing some data, e.g. the reasons why the packages were installed or where the packages were installed from."

If we do not give the user a chance to repair the files, we should rather (with a warning and a log entry) regenerate the files automatically without bothering the user with a manual deletion.

Reply:
Yeah, automatic regeneration would be ideal, and that's the ultimate goal. However, before we implement it, I'd like to resolve https://github.com/rpm-software-management/dnf5/issues/1610 about generating system state from the history database.
Additionally, printing a warning to the user would require a mechanism to store and retrieve non-fatal errors during the `base.setup()` call. Since the library itself can't print errors, this needs to be managed by the DNF5 CLI tool.